server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -86,10 +86,15 @@ public static class Builder extends MetadataFieldMapper.Builder {
             .setMergeValidator(
                 (previous, current, conflicts) -> (previous.value() == current.value()) || (previous.value() && current.value() == false)
             );
+
+        /*
+         * The default mode for TimeSeries is left empty on purpose, so that mapping printings include the synthetic
+         * source mode.
+         */
         private final Parameter<Mode> mode = new Parameter<>(
             "mode",
             true,
-            () -> getIndexMode() == IndexMode.TIME_SERIES ? Mode.SYNTHETIC : null,
+            () -> null,
```

Comment:
I think this is the crux of the issue that this pr is trying to fix.
The default (which is `null`) is used for stored source mode. So if source mode=stored is provided we see that it is different than the default and include it in the mapping. But the default for tsdb is synthetic, so if synthetic is provided this is seen as default and source mode then isn't serialised in the mapping.

Reply:
Right, that's my understanding too. Any concerns with this fix?

Reply:
Nope this is good.